### PR TITLE
Extract a function for parsing op DEPTH_TO_SPACE

### PR DESCRIPTION
**Pull request by rkuester**
Extract the parsing of op parameters from the flatbuffer out of a
switch statement case, into a standalone function which can be
called by the micro op resolver.

This PR is part of the work to port operator DEPTH_TO_SPACE
from lite to micro, as tracked in issue #46025.

### CODE REVIEW

**Review by petewarden — APPROVED** [review]

File: tensorflow/lite/core/api/flatbuffer_conversions.cc
Comment (advaitjain) [inline tensorflow/lite/core/api/flatbuffer_conversions.cc:1062]: we do want to keep this TODO as is. If it is triggering some failure in the CI then we'll need to fix the CI. Can you please change it back to TODO(b/): and then I'll take a look at what the CI error is.

**Review by advaitjain — CHANGES_REQUESTED** [review]

**Review by advaitjain — APPROVED** [review]

**Comment by advaitjain** [comment]
All checks are passing.

Merging a PR needs two approvals from a Googler so we'll need to wait until @petewarden can approve the change in the internal review system before this can be merged.

**Review by petewarden — APPROVED** [review]

**Comment by rkuester** [comment]
@advaitjain not sure what internal check this one is failing.

**Comment by advaitjain** [comment]
The failing tests are internal, unrelated and currently flaky. I have set the internal change to submit while ignoring these tests. This PR should get merged shortly.

**Pull request merged**
